File: loadercomplex-fabric/src/ver122/java/com/enderzombi102/loadercomplex/fabric12/impl/item/FabricItemStack.java
package com.enderzombi102.loadercomplex.fabric12.impl.item;

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack;
import org.jetbrains.annotations.NotNull;

public class FabricItemStack implements ItemStack {

	private final net.minecraft.item.ItemStack stack;

	public FabricItemStack( net.minecraft.item.ItemStack stack ) {
		this.stack = stack;
	}

	public net.minecraft.item.@NotNull ItemStack getStack() {
		return this.stack;
	}

	@Override
	public String getDisplayName() {
		return this.stack.getHoverName();
	}

	@Override
	public void setDisplayName( String newName ) {
		this.stack.setHoverName( newName );
	}

	@Override
	public Object getType() {
		return this.stack.getItem();
	}

	@Override
	public int getAmount() {
		return this.stack.getSize();
	}

	@Override
	public void setAmount( int amount ) {
		this.stack.setSize( amount );
	}

	@Override
	public void setDamage( int damage ) {
		this.stack.setDamage( damage );
	}

	@Override
	public int getDamage() {
		return this.stack.getDamage();
	}

	@Override
	public int getMaxDurability() {
		return this.stack.getMaxDamage();
	}

	@Override
	public int getMaxStackSize() {
		return this.stack.getMaxSize();
	}
}

File: loadercomplex-fabric/src/ver122/kotlin/com/enderzombi102/loadercomplex/fabric122/impl/item/FabricItemStack.kt
package com.enderzombi102.loadercomplex.fabric122.impl.item

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack
import net.minecraft.item.ItemStack as McItemStack


class FabricItemStack( private val stack: McItemStack ) : ItemStack {
	override fun getStack(): McItemStack =
		this.stack

	override fun getDisplayName(): String? =
		stack.hoverName

	override fun setDisplayName( newName: String ) {
		stack.setHoverName( newName )
	}

	override fun getType(): Any =
		stack.item

	override fun getAmount(): Int =
		stack.size

	override fun setAmount( amount: Int ) {
		stack.size = amount
	}

	override fun setDamage( damage: Int ) {
		stack.damage = damage
	}

	override fun getDamage(): Int =
		stack.damage

	override fun getMaxDurability(): Int =
		stack.maxDamage

	override fun getMaxStackSize(): Int =
		stack.maxSize
}
